src/Controller/TaskController.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\Task;
+use App\Form\TaskType;
+use App\Repository\TaskRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Attribute\Route;
+
+class TaskController extends AbstractController
+{
+
+    public function __construct(private EntityManagerInterface $em)
+    {
+    }
+
+    #[Route('/task', name: 'app_task',  methods: ["GET"])]
+    public function index(TaskRepository $taskrepo): Response
+    {
+        $this->denyAccessUnlessGranted("ROLE_USER");
+
+        $tasks = $taskrepo->findAll();
+
+        return $this->render('task/task.html.twig', [
+            'tasks' => $tasks
+        ]);
+    }
+
+    #[Route('/task/edit/{id}', name: 'app_task_edit',  methods: ["GET", "POST"])]
+    public function edit(Task $task, Request $request): Response
+    {
+        $this->denyAccessUnlessGranted("ROLE_USER");
+
+        $form = $this->createForm(TaskType::class, $task);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $data = $form->getData();
+            $task->setTaskname($data->getTaskname());
+            $task->setDescription($data->getDescription());
+            $task->setPoints($data->getPoints());
+            $this->em->persist($task);
+            $this->em->flush();
+            $this->addFlash('success', 'Tâche modifiée');
+
+            return $this->redirectToRoute('app_task');
+        }
+
+        return $this->render('task/edit.html.twig', [
+            'form' => $form,
+            'task' => $task
+        ]);
+    }
+
+    #[Route('/task/delete/{id}', name: 'app_task_delete',  methods: ["GET", "POST"])]
+    public function delete(TaskRepository $taskrepo, int $id): Response
+    {
+        $this->denyAccessUnlessGranted("ROLE_USER");
+
+        $task = $taskrepo->find($id);
```

Comment:
Ici `$task` peut être vide, tu devrais vérifier que c'est une instance de Task ou que c'est non null avant de faire `$this->em->remove($task);`

Reply:
J'ai modifié ma méthode en mettant le paramConverter pour le delete comme indiqué plus haut, donc la question ne se pose plus visiblement (vu avec Erwann).